inc/pantheon-updates.php

```diff
@@ -1,119 +1,186 @@
+		$version = '<span style="font-weight: normal;">(' . $wp_version . ')</span>';
+		$paragraph_style = esc_attr( 'font-size: 14px;' );
+		// Translators: %s is the current WordPress version.
+		$notice_message = sprintf( __( '<strong>You are using a development version of WordPress.</strong> %s', 'pantheon-systems' ), $version );
+		// If we're on the Updates page, add a note about the Beta Tester plugin.
+		if ( 'update-core' === $screen->id || 'update-core-network' === $screen->id ) {
+			$notice_message .= '<br /><span style="font-weight: normal;">';
+			$notice_message .= __( 'You are responsible for keeping WordPress up-to-date. Pantheon updates to WordPress will not appear in the dashboard as long as you\'re using a pre-release version. If you are using the Beta Tester plugin, you must have your site in SFTP mode to get the latest updates to your Pantheon Dev environment.', 'pantheon-systems' );
+		}
+	}
+
+	ob_start();
+	?>
+	<div class="<?php echo esc_attr( $div_class ); ?>" style="<?php echo esc_attr( $div_style ); ?>">
+		<p style="<?php echo esc_attr( $paragraph_style ); ?>">
+			<?php echo wp_kses_post( $notice_message ); ?>
+		</p>
+		<?php if ( ! _pantheon_is_wordpress_core_prerelease() ) : ?>
+			<?php echo wp_kses_post( $upstream_help_message ); ?>
+			<br />
+			<?php echo wp_kses_post( $update_help ); ?>
+		<?php endif; ?>
+	</div>
+	<?php
+	$notice_html = ob_get_clean();
 	if ( _pantheon_is_wordpress_core_latest() ) {
 		// If a WP core update is not detected, only show the nag on the updates page.
-		$screen = get_current_screen(); 
-		if ( 'update-core' === $screen->id || 'update-core-network' === $screen->id ) { ?>
-			<div class="<?php echo $div_class; ?>" style="<?php echo $div_style; ?>">
-				<p style="<?php echo $paragraph_style; ?>">
-					<?php echo $notice_message; ?>
-				</p>
-				<?php echo $upstream_help_message; ?>
-				<br />
-				<?php echo $update_help; ?>
-			</div>
-			<?php
+		if ( 'update-core' === $screen->id || 'update-core-network' === $screen->id ) {
+			// Escaping is handled above when we're buffering the output, so we can ignore it here.
+			echo $notice_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		}
 	} else {
-		// If WP core is out of date, alter the message and show the nag everywhere.
-		// Translators: %s is a URL to the user's Pantheon Dashboard.
-		$notice_message = wp_kses_post( sprintf( __( 'A new WordPress update is available! Please update from <a href="%s">your Pantheon dashboard</a>.', 'pantheon-systems' ), 'https://dashboard.pantheon.io/sites/' . $_ENV['PANTHEON_SITE'] ) );; ?>
-		<div class="<?php echo $div_class; ?>" style="<?php echo $div_style; ?>">
-			<p style="<?php echo $paragraph_style; ?>">
-				<?php echo $notice_message; ?>
-			</p>
-			<?php echo $upstream_help_message; ?>
-			<br />
-			<?php echo $update_help; ?>
-		</div>
-		<?php
+		echo $notice_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 	}
```

Comment:
I can't suggest here due to diff, but I think this could be
```
// If a WP core update is not detected, only show the nag on the updates page.
if ( ! _pantheon_is_wordpress_core_latest() || 'update-core' === $screen->id || 'update-core-network' === $screen->id ) {
	// Escaping is handled above when we're buffering the output, so we can ignore it here.
	echo $notice_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
}
```

Reply:
I think that's a slightly different path, but I tweaked the conditional to simplify it further and combine the checks.